File: GO/armstrong.go
package main

import "fmt"

func main() {
	var num int = 0
	var rem int = 0
	var res int = 0
	var temp int = 0

	fmt.Print("Enter Number: ")
	fmt.Scanf("%d", &num)

	temp = num
MyLbl:

	rem = num % 10
	res = res + (rem * rem * rem)
	num = num / 10

	if num > 0 {
		goto MyLbl
	}

	if temp == res {
		fmt.Printf("Number is armstrong")
	} else {
		fmt.Printf("Number is not armstrong")
	}
}

File: C/armstrong.c
#include <stdio.h>
#include <conio.h>
#include <math.h>

int main()
{
    int num, var, rem, sum = 0, a = 0;

    printf("Please enter an integer:");
    scanf("%d", &num);

    var = num;

    while (var != 0)
    {
        var = var / 10;
        ++a;
    }

    var = num;

    while (var > 0)
    {
        rem = var % 10;
        sum = sum + pow(rem, a);
        var = var / 10;
    }

    if (sum == num)
    {
        printf("%d is an Armstrong number \n", num);
    }
    else
    {
        printf("%d is not an Armstrong number \n", num);
    }
    return 0;
}